cdap-common/src/main/java/io/cdap/cdap/common/service/AbstractRetryableScheduledService.java

```diff
@@ -160,15 +160,19 @@ protected final void runOneIteration() throws Exception {
         delayMillis = runTask();
         nonFailureStartTime = 0L;
         failureCount = 0;
-      } catch (Exception e) {
-        logTaskFailure(e);
+      } catch (Throwable t) {
+        logTaskFailure(t);
+        if (!(t instanceof Exception)) {
+          throw t;
+        }
+        Exception e = (Exception) t;
         if (!shouldRetry(e)) {
```

Comment:
we can use `t` here as well

Reply:
Can't. The `shouldRetry` only take `Exception`. Technically all "Error" are not retryable.